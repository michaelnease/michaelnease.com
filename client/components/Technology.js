import {
  TechnologyStyled,
  Texture,
  FrameworkList,
  ListItem,
  Landing,
  Link
} from "./styles";
import { technology } from "../static/data/technology";

const Technology = () => {
  return (
    <TechnologyStyled>
      <Link name="technology" />
      <Texture>
        <FrameworkList>
          {technology.map(({ id, name }) => (
            <ListItem key={id} name={name} />
          ))}
        </FrameworkList>
      </Texture>
    </TechnologyStyled>
  );
};

export default Technology;
