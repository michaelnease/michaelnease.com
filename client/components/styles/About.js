import styled from "styled-components";

export const AboutStyled = styled.section`
  height: 80rem;
  position: relative;
  background: url("/static/images/bio/bio.gif") no-repeat center center;
  background-size: cover;
  border-bottom: 10px solid ${props => props.theme.colors.black};
  @media (max-width: 700px) {
    height: 100rem;
  }
`;

export const Text = styled.div`
  width: 60rem;
  p {
    text-align: justify;
    font-weight: 300;
    font-size: 1.7rem;
    margin-bottom: 20px;
  }
  @media (max-width: 700px) {
    padding: 3rem;
  }
`;

export const Rex = styled.img`
  box-shadow: ${props => props.theme.effects.bs};
  border: 1px solid ${props => props.theme.effects.black};
  float: left;
  width: 20rem;
  margin: ${props => props.theme.spacing.margin};
  position: relative;
  top: 10px;
`;
