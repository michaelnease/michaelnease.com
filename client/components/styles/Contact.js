import styled from "styled-components";

export const Form = styled.form`
  display: flex;
  position: relative;
  align-content: center;
  justify-content: center;
  background-color: ${props => props.theme.colors.darkGray};
  border-bottom: 10px solid ${props => props.theme.colors.black};
  padding: 3rem;
`;

export const Inputs = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  width: 80rem;
  grid-gap: 1.5rem;

  input,
  textarea {
    border: none;
    background-color: ${props => props.theme.colors.gray};
    padding-left: 3rem;
    font-size: 1.5rem;
    box-shadow: ${props => props.theme.effects.bs};
  }

  input {
    height: 5rem;
  }

  textarea {
    padding: 3rem;
    grid-column: span 2;
  }
`;
