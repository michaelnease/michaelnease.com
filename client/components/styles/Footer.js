import styled from "styled-components";

export const Foot = styled.footer`
  padding: 3rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 0 10rem;
  p {
    color: ${props => props.theme.colors.gray};
  }

  @media (max-width: 700px) {
    padding: 1rem;
    margin: 0;
  }
`;
