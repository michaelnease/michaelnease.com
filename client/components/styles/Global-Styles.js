import { createGlobalStyle } from "styled-components";

const GlobalStyles = createGlobalStyle`
  @font-face {
    font-family: 'Roboto';
    src: url('/static/fonts/Roboto-Regular.ttf') format('truetype');
    font-weight: normal;
    font-style: normal;
  }

  @font-face {
    font-family: 'Roboto';
    src: url('/static/fonts/Roboto-Light.ttf') format('truetype');
    font-weight: 300;
    font-style: normal;
  }

  html {
    box-sizing: border-box;
    font-size: 10px;
  }

  *, *:before, *:after {
    box-sizing: inherit;
  }

  body, h1, h2, h3, h4, h5, h6, p, ol, ul {
    margin: 0;
    padding: 0;
    font-weight: normal;
  }

  ol, ul {
    list-style: none;
  }

  img {
    max-width: 100%;
    height: auto;
  }

  body {
    padding: 0;
    margin: 0;
    font-size: 1.5rem;
    line-height: 2;
    font-family: "Roboto", Arial, Helvetica, sans-serif;
  }

  a {
    text-decoration: none;
  }
  
  button { font-family: "Roboto", Arial, Helvetica, sans-serif; }

  button:focus {
    outline:0;
  }

  textarea:focus, input:focus{
    outline: none;
  }
`;

export default GlobalStyles;
