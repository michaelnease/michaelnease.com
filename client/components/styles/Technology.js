import styled from "styled-components";
import Item from "../Item";
import List from "../List";

export const TechnologyStyled = styled.section`
  position: relative;
  background: url("/static/images/technology/technology.gif") no-repeat center
    center;
  background-size: cover;
  border-bottom: 10px solid ${props => props.theme.colors.black};
`;

export const FrameworkList = styled(List)`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 48rem;

  & > div {
    width: 70rem;
    height: 40rem;
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-gap: 15px;

    @media (max-width: 700px) {
      width: 30rem;
    }
  }
`;

export const ListItem = styled(Item)`
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: ${props => props.theme.colors.black};
  box-shadow: ${props => props.theme.effects.bs};
  & span {
    color: ${props => props.theme.colors.white};
  }
`;
