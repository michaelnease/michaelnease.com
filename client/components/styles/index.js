export * from "./Header";
export * from "./Quotes";
export * from "./About";
export * from "./Technology";
export * from "./Portfolio";
export * from "./Contact";
export * from "./Footer";
export * from "./Send-Message-Button";

export * from "./Texture";
