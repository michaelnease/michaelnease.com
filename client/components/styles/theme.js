export const theme = {
  colors: {
    black: "#141414",
    white: "#fff",
    gray: "#e0e0e0",
    darkGray: "#a9a9a9"
  },
  spacing: {
    margin: "15px",
    padding: "15px"
  },
  effects: {
    bs: "-5px 5px 5px 0px rgba(0, 0, 0, 0.2)"
  },
  bar: {
    height: "60px"
  }
};
