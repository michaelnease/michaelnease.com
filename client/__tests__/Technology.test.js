import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { render } from "react-testing-library";
import "jest-styled-components";
import { ThemeProvider } from "styled-components";

import { theme } from "../components/styles/theme";
import Technology from "../components/Technology";

configure({ adapter: new Adapter() });

describe("<Technology />", () => {
  const wrapper = shallow(<Technology />);

  it("Should render 12 <ListItem /> elements", () => {
    expect(wrapper.find("ListItem").length).toBe(12);
  });

  it("Should render <ListItem /> elements", () => {
    expect(
      wrapper
        .find("FrameworkList")
        .childAt(0)
        .prop("name")
    ).toBe("React");
    expect(
      wrapper
        .find("FrameworkList")
        .childAt(1)
        .prop("name")
    ).toBe("Redux");
    expect(
      wrapper
        .find("FrameworkList")
        .childAt(2)
        .prop("name")
    ).toBe("Javascript");
    expect(
      wrapper
        .find("FrameworkList")
        .childAt(3)
        .prop("name")
    ).toBe("CSS");
    expect(
      wrapper
        .find("FrameworkList")
        .childAt(4)
        .prop("name")
    ).toBe("HTML");
    expect(
      wrapper
        .find("FrameworkList")
        .childAt(5)
        .prop("name")
    ).toBe("Node");
    expect(
      wrapper
        .find("FrameworkList")
        .childAt(6)
        .prop("name")
    ).toBe("Express");
    expect(
      wrapper
        .find("FrameworkList")
        .childAt(7)
        .prop("name")
    ).toBe("MongoDB");
    expect(
      wrapper
        .find("FrameworkList")
        .childAt(8)
        .prop("name")
    ).toBe("Docker");
    expect(
      wrapper
        .find("FrameworkList")
        .childAt(9)
        .prop("name")
    ).toBe("Git");
    expect(
      wrapper
        .find("FrameworkList")
        .childAt(10)
        .prop("name")
    ).toBe("Ruby");
    expect(
      wrapper
        .find("FrameworkList")
        .childAt(11)
        .prop("name")
    ).toBe("AWS");
  });

  it("Should match Snapshot", () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <Technology />
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
