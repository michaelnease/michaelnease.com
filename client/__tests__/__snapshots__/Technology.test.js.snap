// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Technology /> Should match Snapshot 1`] = `
<DocumentFragment>
  .c1 {
  position: absolute;
  top: -60px;
}

.c0 {
  position: relative;
  background: url("/static/images/technology/technology.gif") no-repeat center center;
  background-size: cover;
  border-bottom: 10px solid #141414;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 48rem;
}

.c3 > div {
  width: 70rem;
  height: 40rem;
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 15px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #141414;
  box-shadow: -5px 5px 5px 0px rgba(0,0,0,0.2);
}

.c4 span {
  color: #fff;
}

.c2 {
  background-image: url("/static/images/common/texture.png");
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (max-width:700px) {
  .c3 > div {
    width: 30rem;
  }
}

<section
    class="c0"
  >
    <a
      class="c1"
      name="technology"
    />
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div>
          <div
            class="c4"
          >
            <span>
              React
            </span>
          </div>
          <div
            class="c4"
          >
            <span>
              Redux
            </span>
          </div>
          <div
            class="c4"
          >
            <span>
              Javascript
            </span>
          </div>
          <div
            class="c4"
          >
            <span>
              CSS
            </span>
          </div>
          <div
            class="c4"
          >
            <span>
              HTML
            </span>
          </div>
          <div
            class="c4"
          >
            <span>
              Node
            </span>
          </div>
          <div
            class="c4"
          >
            <span>
              Express
            </span>
          </div>
          <div
            class="c4"
          >
            <span>
              MongoDB
            </span>
          </div>
          <div
            class="c4"
          >
            <span>
              Docker
            </span>
          </div>
          <div
            class="c4"
          >
            <span>
              Git
            </span>
          </div>
          <div
            class="c4"
          >
            <span>
              Ruby
            </span>
          </div>
          <div
            class="c4"
          >
            <span>
              AWS
            </span>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
